workflow/controller/controller_test.go

```diff
@@ -1181,6 +1181,51 @@ spec:
 	assert.Len(t, pods.Items, 0)
 }
 
+func TestPodCleaupPatch(t *testing.T) {
+	wfc := &WorkflowController{}
+
+	pod := &apiv1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels:          map[string]string{common.LabelKeyCompleted: "false"},
+			Finalizers:      []string{common.FinalizerPodStatus},
+			ResourceVersion: "123456",
+		},
+	}
+
+	os.Setenv(common.EnvVarPodStatusCaptureFinalizer, "true")
```

Comment:
```suggestion
	t.Setenv(common.EnvVarPodStatusCaptureFinalizer, "true")
```

`t.Setenv` has a cleanup at the end. 

You solved the E2E failures it seems like, but I was guessing at places that could cause it and lack of cleanup was one possibility I thought of.